wfe/web-front-end.go

```diff
@@ -467,69 +428,41 @@ func (wfe *WebFrontEndImpl) verifyPOST(logEvent *requestEvent, request *http.Req
 	if err != nil {
 		wfe.stats.Inc("WFE.Errors.UnparsableJWSPayload", 1, 1.0)
 		logEvent.AddError("unable to JSON parse resource from JWS payload: %s", err)
-		puberr := core.SignatureValidationError("Request payload did not parse as JSON")
-		return nil, nil, reg, puberr
+		return nil, nil, reg, probs.Malformed("Request payload did not parse as JSON")
 	}
 	if parsedRequest.Resource == "" {
 		wfe.stats.Inc("WFE.Errors.NoResourceInJWSPayload", 1, 1.0)
-		logEvent.AddError("JWS request payload does not specifiy a resource")
-		err = core.MalformedRequestError("Request payload does not specify a resource")
-		return nil, nil, reg, err
+		logEvent.AddError("JWS request payload does not specify a resource")
+		return nil, nil, reg, probs.Malformed("Request payload does not specify a resource")
 	} else if resource != core.AcmeResource(parsedRequest.Resource) {
 		wfe.stats.Inc("WFE.Errors.MismatchedResourceInJWSPayload", 1, 1.0)
 		logEvent.AddError("JWS request payload does not match resource")
-		err = core.MalformedRequestError(fmt.Sprintf("JWS resource payload does not match the HTTP resource: %s != %s", parsedRequest.Resource, resource))
-		return nil, nil, reg, err
+		return nil, nil, reg, probs.Malformed("JWS resource payload does not match the HTTP resource: %s != %s", parsedRequest.Resource, resource)
 	}
 
 	return []byte(payload), key, reg, nil
 }
 
-// Notify the client of an error condition and log it for audit purposes.
-func (wfe *WebFrontEndImpl) sendError(response http.ResponseWriter, logEvent *requestEvent, msg string, detail error, code int) {
-	problem := probs.ProblemDetails{Detail: msg, HTTPStatus: code}
-	switch code {
-	case http.StatusPreconditionFailed:
-		fallthrough
-	case http.StatusForbidden:
-		problem.Type = probs.UnauthorizedProblem
-	case http.StatusConflict:
-		fallthrough
-	case http.StatusMethodNotAllowed:
-		fallthrough
-	case http.StatusNotFound:
-		fallthrough
-	case http.StatusBadRequest:
-		fallthrough
-	case http.StatusLengthRequired:
-		problem.Type = probs.MalformedProblem
-	case StatusRateLimited:
-		problem.Type = probs.RateLimitedProblem
-	case statusBadNonce:
-		problem.Type = probs.BadNonceProblem
-		problem.HTTPStatus = http.StatusBadRequest
-		code = http.StatusBadRequest
-	default: // Either http.StatusInternalServerError or an unexpected code
-		problem.Type = probs.ServerInternalProblem
-	}
+// sendError sends an error response represented by the given ProblemDetails,
+// and, if the ProblemDetails.Type is ServerInternalProblem, audit logs the
+// internal ierr.
+func (wfe *WebFrontEndImpl) sendError(response http.ResponseWriter, logEvent *requestEvent, prob *probs.ProblemDetails, ierr error) {
+	code := probs.ProblemDetailsToStatusCode(prob)
 
 	// Record details to the log event
-	logEvent.AddError(msg)
+	logEvent.AddError(fmt.Sprintf("%s :: %s :: %d", prob.Type, prob.Detail, prob.HTTPStatus))
```

Comment:
Looks like adding the status here is new, right? Let's put it in front, since it's the easiest thing to visually scan for.